db/migrate/20201124063738_create_addresses.rb

```diff
@@ -0,0 +1,12 @@
+
+class CreateAddresses < ActiveRecord::Migration[5.2]
+  def change
+    create_table :addresses do |t|
+      t.references :ems, :type => :bigint, :index => true, :references => :ext_management_system
+      t.string :ems_ref
+      t.references :physical_storage, :type => :bigint, :index => true
```

Comment:
An address belongs to a `physical_storage_consumer`, not a `physical_storage`, correct? If that's the case, creating a reference to `physical_storage` is duplicate date, because the storage is determined by the consumer.